docs/setup/upgrade.asciidoc

```diff
@@ -44,13 +44,12 @@ a|
 [[upgrade-before-you-begin]]
 === Before you begin
 
-WARNING: {kib} automatically runs upgrade migrations when required. To roll back to an earlier version in case of an upgrade failure, you **must** have a {ref}/snapshot-restore.html[backup snapshot] available. This snapshot must include the `kibana` feature state or all `kibana*` indices. For more information see <<upgrade-migrations, upgrade migrations>>.
+WARNING: {kib} automatically runs upgrade migrations when required. To roll back to an earlier version in case of an upgrade failure, you **must** have a {ref}/snapshot-restore.html[backup snapshot] available. This snapshot must include the `kibana` feature state. For more information see <<upgrade-migrations, upgrade migrations>>.
 
 Before you upgrade {kib}:
 
 * Consult the <<breaking-changes,breaking changes>>.
-* {ref}/snapshots-take-snapshot.html[Take a snapshot] of your data. To roll back to an earlier version, the snapshot must include the `kibana` feature state or all `.kibana*` indices. 
-* Although not a requirement for rollbacks, we recommend taking a snapshot of all {kib} indices created by the plugins you use such as the `.reporting*` indices created by the reporting plugin.  
```

Comment:
Reporting isn't in a separate plugin anymore so I removed this bullet. I don't think plugins are as popular anymore, but let me know if we still want to include an updated variation of this.

Reply:
@rudolf

Reply:
this was meant as a catch-all for several indices like `.kibana_security_session` `.kibana-event-log-8.0.0` I assume the kibana feature will include all indices matching `.kibana*`.

We also have some other indices like `.apm-agent-configuration` and `.apm-custom-link` that are system indices, but might not be included in the kibana feature?

I couldn't find the PR that introduced the feature snapshot, if you happen to have a link that would be useful, otherwise I could ask.

I think it's much better to hide this complexity from a user under a "feature snapshot" so happy with these docs changes, but I just want to make sure we've covered all our bases.

Reply:
I tried the following but got an error
```
curl -X GET "elastic:changeme@localhost:9200/_snapshottable_features" -H 'Content-Type: application/json' 
{"error":{"root_cause":[{"type":"invalid_index_name_exception","reason":"Invalid index name [_snapshottable_features], must not start with '_'.","index_uuid":"_na_","index":"_snapshottable_features"}],"type":"invalid_index_name_exception","reason":"Invalid index name [_snapshottable_features], must not start with '_'.","index_uuid":"_na_","index":"_snapshottable_features"},"status":400}
```

Reply:
@rudolf The route to get the list of features is [`GET _features`](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-features-api.html). It was changed from `_snapshottable_features` late in development, so if we've missed somewhere in the docs that you found it, please let me know so we can get it updated.

Reply:
Thanks, I found _snapshottable_features in some PR's but couldn't find any mentions in our docs. Do we have any internal docs explaining which system indices are included in the different features?

Reply:
> We also have some other indices like `.apm-agent-configuration` and `.apm-custom-link` that are system indices, but might not be included in the kibana feature?

Those indices are included. You can see the system indices and related index patterns for the `kibana` feature state here:
https://github.com/elastic/elasticsearch/blob/master/modules/kibana/src/main/java/org/elasticsearch/kibana/KibanaPlugin.java

I verified by using get snapshot API on a 7.16 snapshot created using the default ESS snapshot policy. Here's the relevant bit of the response:

```
{
  "snapshots" : [
    {
      ...
      "feature_states" : [
        ...,
        {
          "feature_name" : "kibana",
          "indices" : [
            ".apm-custom-link",
            ".kibana_task_manager_7.16.0_001",
            ".apm-agent-configuration",
            ".kibana_7.16.0_001",
            ".kibana_security_session_1"
          ]
        }
      ]
    }
  ],
  "total" : 1,
  "remaining" : 0
}
 ```